newsfragments/2406.bugfix.rst

```diff
@@ -0,0 +1,2 @@
+In the Worker's commitment tracker, the logic to check for what's the current period should happen before the logic to track pending transactions.
```

Comment:
At minimum - I'd change should -> must...

```suggestion
 In the Worker's commitment tracker, the logic to check for what's the current period must happen before the logic to track pending transactions.
```

... However this sounds more like a detailed description than it does a news headline.  Here is a more terse example:

```suggestion
Ensure replacement transactions can only be issued once per period.
```

Reply:
✅